blaze/compute/numpy.py

```diff
@@ -122,12 +150,12 @@ def compute_up(t, x, **kwargs):
 
 @dispatch(Not, np.ndarray)
 def compute_up(t, x, **kwargs):
-    return ~x
+    return np.logical_not(x)
```

Comment:
Does our `Not` only work with bools? This gives different results for dtypes other than `bool`

```python
In [2]: ~np.array([1, 2, 3])
Out[2]: array([-2, -3, -4])

In [3]: np.logical_not(np.array([1, 2, 3]))
Out[3]: array([False, False, False], dtype=bool)
```